exotations/exotica_core_task_maps/src/eff_box.cpp

```diff
@@ -71,14 +71,16 @@ void EffBox::Update(Eigen::VectorXdRefConst x, Eigen::VectorXdRef phi, Eigen::Ma
     if (debug_ && Server::IsRos()) PublishObjectsAsMarkerArray();
 }
 
-Eigen::VectorXd EffBox::GetLowerLimit() const
+Eigen::Vector3d EffBox::GetLowerLimit(const int eff_id) const
 {
-    return eff_lower_;
+    if (eff_id < 0 || eff_id >= n_effs_) ThrowNamed("Given eff_id (" << eff_id << ") is out of range [0, " << n_effs_ << ")!");
+    return eff_lower_.segment<3>(3 * eff_id);
 }
 
-Eigen::VectorXd EffBox::GetUpperLimit() const
+Eigen::Vector3d EffBox::GetUpperLimit(const int eff_id) const
 {
-    return eff_upper_;
+    if (eff_id < 0 || eff_id >= n_effs_) ThrowNamed("Given eff_id (" << eff_id << ") is out of range [0, " << n_effs_ << ")!");
```

Comment:
I think you need to check against n_effs_ - 1.

Reply:
Let there be two end-effectors that you are constraining using `EffBox`, then `n_effs_ = 2` and the only possible values for `eff_id` is `0` or `1`. 

If I give an `eff_id` from the set `{..., -3, -2, -1}` then 
```
eff_id < 0 || eff_id >= n_effs_ -> True
   True    or    False
```
If I give an `eff_id` from the set `{2, 3, 4, ...}` then
```
eff_id < 0 || eff_id >= n_effs_ -> True
   False   or    True      
```
If I give an `eff_id` from the set `{0, 1}` then 
```
eff_id < 0 || eff_id >= n_effs_ -> False
   False   or    False
```

Reply:
No need for such a mathematical explanation ;-). My argument was `n_end_effs_` is the number of end-effectors. So when you have `2` (ids `0` and `1`), `n_end_effs_` is 2.

Nevermind, realised I didn't correct interpret the `>=` in this case.